client/my-sites/plugins/plugin-details.jsx

```diff
@@ -0,0 +1,440 @@
+import { recordGoogleEvent } from 'calypso/state/analytics/actions';
+import {
+	getPluginOnSite,
+	getPluginOnSites,
+	getSiteObjectsWithPlugin,
+	getSitesWithoutPlugin,
+	isRequestingForSites,
+} from 'calypso/state/plugins/installed/selectors';
+import { removePluginStatuses } from 'calypso/state/plugins/installed/status/actions';
+import { fetchPluginData as wporgFetchPluginData } from 'calypso/state/plugins/wporg/actions';
+import {
+	isFetching as isWporgPluginFetching,
+	isFetched as isWporgPluginFetched,
+	getPlugin as getWporgPlugin,
+} from 'calypso/state/plugins/wporg/selectors';
+import { canCurrentUser } from 'calypso/state/selectors/can-current-user';
+import canCurrentUserManagePlugins from 'calypso/state/selectors/can-current-user-manage-plugins';
+import getSelectedOrAllSitesWithPlugins from 'calypso/state/selectors/get-selected-or-all-sites-with-plugins';
+import {
+	isJetpackSite as checkJetpackSite,
+	isRequestingSites as checkRequestingSites,
+} from 'calypso/state/sites/selectors';
+import { getSelectedSite, getSelectedSiteId } from 'calypso/state/ui/selectors';
+import NoPermissionsError from './no-permissions-error';
+
+function PluginDetails( props ) {
+	const dispatch = useDispatch();
+	const moment = useLocalizedMoment();
+	const translate = useTranslate();
+
+	const selectedSiteId = useSelector( getSelectedSiteId );
+	const sites = useSelector( getSelectedOrAllSitesWithPlugins );
+	const siteIds = [ ...new Set( siteObjectsToSiteIds( sites ) ) ];
+
+	const plugin = useSelector( ( state ) => getPluginOnSites( state, siteIds, props.pluginSlug ) );
+	const wporgPlugin = useSelector( ( state ) => getWporgPlugin( state, props.pluginSlug ) );
+	const isFetching = useSelector( ( state ) => isWporgPluginFetching( state, props.pluginSlug ) );
+	const isFetched = useSelector( ( state ) => isWporgPluginFetched( state, props.pluginSlug ) );
+	const selectedSite = useSelector( getSelectedSite );
+	const isJetpackSite = useSelector(
+		( state ) => selectedSiteId && checkJetpackSite( state, selectedSiteId )
+	);
+	const isRequestingSites = useSelector( checkRequestingSites );
+	const requestingPluginsForSites = useSelector( ( state ) =>
+		isRequestingForSites( state, siteIds )
+	);
+	const sitePlugin = useSelector(
+		( state ) => selectedSiteId && getPluginOnSite( state, selectedSiteId, props.pluginSlug )
+	);
+	const userCanManagePlugins = useSelector( ( state ) =>
+		selectedSiteId
+			? canCurrentUser( state, selectedSiteId, 'manage_options' )
+			: canCurrentUserManagePlugins( state )
+	);
+
+	const isWpcom = selectedSite && ! isJetpackSite;
+	const analyticsPath = selectedSite ? '/plugins/:plugin/:site' : '/plugins/:plugin';
+
+	const isPluginInstalledOnsite = ! requestingPluginsForSites ? !! sitePlugin : null;
+
+	const fullPlugin = {
+		...plugin,
+		...wporgPlugin,
+	};
+
+	useEffect( () => {
+		if ( ! isFetched ) {
+			dispatch( wporgFetchPluginData( props.pluginSlug ) );
+		}
+	}, [ isFetched ] );
```

Comment:
is there a reason for not adding `dispatch` and `props.pluginSlug` as dependencies?

Reply:
It was a quick migration from the class component, as you can see [here](https://github.com/Automattic/wp-calypso/blob/5a88fc6a29779919968b6e2f4e469f6e2fccc210/client/my-sites/plugins/plugin.jsx#L47-L52).

That said, I'm fixing on the next commit.